Map keys with dots need to be quoted in Terraform 0.12

I have a variable that accepts a map:

```hcl
variable "method_request_parameters" {
  type        = map(string)
  description = "(Optional) A map of request query string parameters and headers that should be passed to the integration."
}

variable "integration_request_parameters" {
  type        = map(string)
  description = "(Optional) A map of request query string parameters and headers that should be passed to the backend responder."
}
```

I'm trying to set the value in my `terraform.Options` like so:

```go
terraformOptions = &terraform.Options{
	// ...
	Vars: map[string]interface{}{
		"method_request_parameters": map[string]string{
			"method.request.header.Authorization": "false",
		},
		"integration_request_parameters": map[string]string{
			"integration.request.header.Authorization": "method.request.header.Authorization",
		},
	},
}
```

Starting with 0.12, it appears that map keys need to be quoted if they contain periods:

```txt
TestTerraformModule 2019-09-09T17:21:22Z retry.go:72: terraform [apply -input=false -lock=false -auto-approve -var method_request_parameters={method.request.header.Authorization = "false"} -var integration_request_parameters={integration.request.header.Authorization = "method.request.header.Authorization"}]
TestTerraformModule 2019-09-09T17:21:22Z command.go:87: Running command terraform with args [apply -input=false -lock=false -auto-approve -var method_request_parameters={method.request.header.Authorization = "false"} -var integration_request_parameters={integration.request.header.Authorization = "method.request.header.Authorization"}]
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: 
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: Error: Ambiguous attribute key
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: 
TestTerraformModule 2019-09-09T17:21:22Z command.go:158:   on <value for var.integration_request_parameters> line 1:
TestTerraformModule 2019-09-09T17:21:22Z command.go:158:   (source code not available)
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: 
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: If this expression is intended to be a reference, wrap it in parentheses. If
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: it's instead intended as a literal name containing periods, wrap it in quotes
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: to create a string literal.
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: 
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: 
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: Error: Ambiguous attribute key
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: 
TestTerraformModule 2019-09-09T17:21:22Z command.go:158:   on <value for var.method_request_parameters> line 1:
TestTerraformModule 2019-09-09T17:21:22Z command.go:158:   (source code not available)
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: 
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: If this expression is intended to be a reference, wrap it in parentheses. If
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: it's instead intended as a literal name containing periods, wrap it in quotes
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: to create a string literal.
TestTerraformModule 2019-09-09T17:21:22Z command.go:158: 
TestTerraformModule 2019-09-09T17:21:22Z retry.go:80: Returning due to fatal error: FatalError{Underlying: exit status 1}
```

I can quote this manually to work around it, but it would be awesome if terratest did this for you:

```go
terraformOptions = &terraform.Options{
	// ...
	Vars: map[string]interface{}{
		"method_request_parameters": map[string]string{
			`"method.request.header.Authorization"`: "false",
		},
		"integration_request_parameters": map[string]string{
			`"integration.request.header.Authorization"`: "method.request.header.Authorization",
		},
	},
}
```

--- a/modules/terraform/format.go
+++ b/modules/terraform/format.go
@@ -137,7 +137,7 @@ func mapToHclString(m map[string]interface{}) string {
 	keyValuePairs := []string{}
 
 	for key, value := range m {
-		keyValuePair := fmt.Sprintf("%s = %s", key, toHclString(value, true))
+		keyValuePair := fmt.Sprintf(`"%s" = %s`, key, toHclString(value, true))
 		keyValuePairs = append(keyValuePairs, keyValuePair)
 	}
 

--- a/modules/terraform/format_test.go
+++ b/modules/terraform/format_test.go
@@ -19,10 +19,10 @@ func TestFormatTerraformVarsAsArgs(t *testing.T) {
 		{map[string]interface{}{"foo": 123}, []string{"-var", "foo=123"}},
 		{map[string]interface{}{"foo": true}, []string{"-var", "foo=1"}},
 		{map[string]interface{}{"foo": []int{1, 2, 3}}, []string{"-var", "foo=[1, 2, 3]"}},
-		{map[string]interface{}{"foo": map[string]string{"baz": "blah"}}, []string{"-var", "foo={baz = \"blah\"}"}},
+		{map[string]interface{}{"foo": map[string]string{"baz": "blah"}}, []string{"-var", "foo={\"baz\" = \"blah\"}"}},
 		{
 			map[string]interface{}{"str": "bar", "int": -1, "bool": false, "list": []string{"foo", "bar", "baz"}, "map": map[string]int{"foo": 0}},
-			[]string{"-var", "str=bar", "-var", "int=-1", "-var", "bool=0", "-var", "list=[\"foo\", \"bar\", \"baz\"]", "-var", "map={foo = 0}"},
+			[]string{"-var", "str=bar", "-var", "int=-1", "-var", "bool=0", "-var", "list=[\"foo\", \"bar\", \"baz\"]", "-var", "map={\"foo\" = 0}"},
 		},
 	}
 
@@ -62,11 +62,12 @@ func TestMapToHclString(t *testing.T) {
 		expected string
 	}{
 		{map[string]interface{}{}, "{}"},
-		{map[string]interface{}{"key1": "value1"}, "{key1 = \"value1\"}"},
-		{map[string]interface{}{"key1": 123}, "{key1 = 123}"},
-		{map[string]interface{}{"key1": true}, "{key1 = 1}"},
-		{map[string]interface{}{"key1": []int{1, 2, 3}}, "{key1 = [1, 2, 3]}"}, // Any value that isn't a primitive is forced into a string
-		{map[string]interface{}{"key1": "value1", "key2": 0, "key3": false}, "{key1 = \"value1\", key2 = 0, key3 = 0}"},
+		{map[string]interface{}{"key1": "value1"}, "{\"key1\" = \"value1\"}"},
+		{map[string]interface{}{"key1": 123}, "{\"key1\" = 123}"},
+		{map[string]interface{}{"key1": true}, "{\"key1\" = 1}"},
+		{map[string]interface{}{"key1": []int{1, 2, 3}}, "{\"key1\" = [1, 2, 3]}"}, // Any value that isn't a primitive is forced into a string
+		{map[string]interface{}{"key1": "value1", "key2": 0, "key3": false}, "{\"key1\" = \"value1\", \"key2\" = 0, \"key3\" = 0}"},
+		{map[string]interface{}{"key1.a.b.c": "value1"}, "{\"key1.a.b.c\" = \"value1\"}"},
 	}
 
 	for _, testCase := range testCases {
@@ -121,8 +122,8 @@ func TestSliceToHclString(t *testing.T) {
 		{[]interface{}{true}, "[1]"},
 		{[]interface{}{[]int{1, 2, 3}}, "[[1, 2, 3]]"}, // Any value that isn't a primitive is forced into a string
 		{[]interface{}{"foo", 0, false}, "[\"foo\", 0, 0]"},
-		{[]interface{}{map[string]interface{}{"foo": "bar"}}, "[{foo = \"bar\"}]"},
-		{[]interface{}{map[string]interface{}{"foo": "bar"}, map[string]interface{}{"foo": "bar"}}, "[{foo = \"bar\"}, {foo = \"bar\"}]"},
+		{[]interface{}{map[string]interface{}{"foo": "bar"}}, "[{\"foo\" = \"bar\"}]"},
+		{[]interface{}{map[string]interface{}{"foo": "bar"}, map[string]interface{}{"foo": "bar"}}, "[{\"foo\" = \"bar\"}, {\"foo\" = \"bar\"}]"},
 	}
 
 	for _, testCase := range testCases {
@@ -144,8 +145,8 @@ func TestToHclString(t *testing.T) {
 		{true, "1"},
 		{[]int{1, 2, 3}, "[1, 2, 3]"},
 		{[]string{"foo", "bar", "baz"}, "[\"foo\", \"bar\", \"baz\"]"},
-		{map[string]string{"key1": "value1"}, "{key1 = \"value1\"}"},
-		{map[string]int{"key1": 123}, "{key1 = 123}"},
+		{map[string]string{"key1": "value1"}, "{\"key1\" = \"value1\"}"},
+		{map[string]int{"key1": 123}, "{\"key1\" = 123}"},
 	}
 
 	for _, testCase := range testCases {

